roles/network_plugin/cilium/templates/cilium/ds.yml.j2

```
@@ -106,6 +106,7 @@ spec:
         - {{ env_var | to_nice_yaml(indent=2) | indent(10) }}
 {% endfor %}
         lifecycle:
+{% if cilium_version | regex_replace('v') is version('1.14', '<=') %}
```

Cilium no longer requires `cni-install.sh` in version 1.14. Should we consider referencing the following changes to align with the upstream ? see https://github.com/cilium/cilium/blob/b5013e1562dbfe442dd2011a5af46de9095315ff/install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml#L209C10-L209C10, 

```suggestion
{% if cilium_version | regex_replace('v') is version('1.14', '<') %}
```

---

I've noticed there are many differences between our manifests and the upstream ones. We can try to align with the upstream as much as possible, but achieving complete consistency may be challenging.

---

nice catch, ty!

> Cilium no longer requires `cni-install.sh` in version 1.14. Should we consider referencing the following changes to align with the upstream ? see https://github.com/cilium/cilium/blob/b5013e1562dbfe442dd2011a5af46de9095315ff/install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml#L209C10-L209C10,

> I've noticed there are many differences between our manifests and the upstream ones. We can try to align with the upstream as much as possible, but achieving complete consistency may be challenging.

I looked at the one you linked to and that seems to only be used when running Cilium on AWS with elastic network interfaces (ENI) enabled. I can look into adding support for that to cilium here, but I don't have an AWS cluster to test those changes with and not entirely sure what other work would be required with Kubespray to get that functionality. 

Definitely agree that achieving complete consistency will be challenging. Personally I think it makes more sense to ensure functionality works correctly and if there's anything the built-in Cilium configs can't support it can be installed via the custom CNI option